### PR TITLE
Add tasks to the influx cli

### PR DESCRIPTION
**Pull request by lyondhill**
TO facilitate this we created a http client implementation of TaskService. 

Three methods are "not yet implemented". I will be creating separate issues for those methods because the task service testing framework doesn't yet test those functions.

### CODE REVIEW

File: cmd/influx/task.go [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+var taskCreateFlags TaskCreateFlags
+
+func init() {
+	taskCreateCmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create task",
+		Run:   taskCreateF,
+	}
+
+	taskCreateCmd.Flags().StringVarP(&taskCreateFlags.orgID, "org-id", "", "", "id of the organization that owns the task")
```
Comment (goller) [inline cmd/influx/task.go]: Would you add org as a name as well?

File: http/task_service.go [hunk 2]
```diff
@@ -130,12 +135,27 @@ func decodeGetTasksRequest(ctx context.Context, r *http.Request) (*getTasksReque
 func (h *TaskHandler) handlePostTask(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 
+	tok, err := GetToken(r)
```
Comment (goller) [inline http/task_service.go]: authentication_middleware handles this now; use GetAuthorizer instead.

File: http/task_service.go [hunk 3]
```diff
@@ -575,3 +601,260 @@ func decodeRetryRunRequest(ctx context.Context, r *http.Request) (*retryRunReque
 		RunID: i,
 	}, nil
 }
+
+type TaskService struct {
```
Comment (goller) [inline http/task_service.go]: Would you be able to add tests for the TaskService as well?  

The testing package has examples about how to use the service and http together to test client and server together.

Comment (lyondhill) [reply http/task_service.go]: The http.TaskService Implements the platform.TaskService interface (https://github.com/influxdata/platform/blob/master/task.go#L33). This make testing any task service very convenient weather its http, rpc, or the platform adaptor. We created a generic testing framework for all of the things that implement this interface (https://github.com/influxdata/platform/blob/master/task/servicetest/servicetest.go).

File: task/servicetest/servicetest.go [hunk 4]
```diff
@@ -426,6 +431,20 @@ func testTaskConcurrency(t *testing.T, sys *System) {
 	extraWg.Wait()
 }
 
+func creds(t *testing.T, s *System) (orgID, userID platform.ID, token string) {
+	t.Helper()
+
+	if s.CredsFunc == nil {
+		return idGen.ID(), idGen.ID(), idGen.ID().String()
```
Comment (docmerlin) [inline task/servicetest/servicetest.go]: Token is just a stringified snowflake ID? something about that bothers me

File: http/task_service.go (hunk 3 above)
Comment (goller) [reply http/task_service.go]: @lyondhill we did the same in `testing`  Do you have tests for the client in place?

Comment (lyondhill) [reply http/task_service.go]: Yes. They are in http/task_test.go

**Pull request merged**